src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.ApiManagement.tests.ps1

```diff
@@ -1567,5 +1567,190 @@ InModuleScope Arcus.Scripting.ApiManagement {
+            }
+            It "Applying user configuration to Azure API Management with a complex config file is OK" {
+                # Arrange
+                $resourceGroup = "SomeResourceGroup"
+                $serviceName = "SomeServiceName"
+                $configFile = "$PSScriptRoot\Files\ApiManagement\create-azapimanagementuseraccountsfromconfig-config-complex.json"
+                $context = New-Object -TypeName Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models.PsApiManagementContext
+                $stubApiManagement = New-Object -TypeName Microsoft.Azure.Commands.ApiManagement.Models.PsApiManagement
+                $userId = "some-userid"
+
+                Mock Write-Host {}
+                Mock Get-AzApiManagement {
+                    $ResourceGroupName | Should -Be $resourceGroup
+                    $Name | Should -Be $serviceName
+                    return $stubApiManagement } -Verifiable
+                Mock New-AzApiManagementContext {
+                    $ResourceGroupName | Should -Be $resourceGroup
+                    $ServiceName | Should -Be $serviceName
+                    return $context } -Verifiable
+                Mock Get-AzCachedAccessToken -MockWith {
+                    return @{
+                        SubscriptionId = "123456"
+                        AccessToken = "accessToken"
+                    }
+                } -Verifiable
+                Mock Create-AzApiManagementUserAccount -MockWith {
+                    return $userId
+                } -Verifiable
+                Mock New-AzApiManagementGroup {} -Verifiable
+                Mock Add-AzApiManagementUserToGroup {} -Verifiable
+                Mock New-AzApiManagementSubscription {} -Verifiable
+
+                # Act
+                Create-AzApiManagementUserAccountsFromConfig -ResourceGroupName $resourceGroup -ServiceName $serviceName -ConfigurationFile $configFile
+                
+                #Assert
+                Assert-VerifiableMock
+                Assert-MockCalled Get-AzApiManagement -Times 1
+                Assert-MockCalled New-AzApiManagementContext -Times 1
+                Assert-MockCalled Get-AzCachedAccessToken -Times 1
+                Assert-MockCalled Create-AzApiManagementUserAccount -Times 1
+                Assert-MockCalled New-AzApiManagementGroup -Exactly 4
+                Assert-MockCalled Add-AzApiManagementUserToGroup -Exactly 4
+                Assert-MockCalled New-AzApiManagementSubscription -Exactly 4
+                Assert-MockCalled Write-Host -Exactly 2 -ParameterFilter { $Object -eq "User configuration has successfully been applied for user with id '$userId' to Azure API Management service '$serviceName' in resource group '$resourceGroup'" }
```

Comment:
Maybe this is a bit too dependent on the script. If we add logging, this would fail, I believe. Maybe we can add some specific assertions on the users/groups that were created/added basedon the `-config-complex.json`? Shame that we do not have a real API management to test against, so I guess verifying that we deserialize the config file correctly and are calling the API management comments correctly with those values is the best we can do.

Reply:
Just did a quick test, if we add logging this does not break the test since we use the ParameterFilter for that specific logging line (we should have 2 Write-Host calls with that specific line indicating that user config was applied). 

Added assertions in [`575e49f` (#422)](https://github.com/arcus-azure/arcus.scripting/pull/422/commits/575e49f1fc96bee58792b19f7d628f71b0386a49)